src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -347,18 +348,17 @@ function BaseSelectionList<TItem extends User | RadioItem>(
         }, [shouldShowTextInput]),
     );
 
+    const prevTextInputValue = usePrevious(textInputValue);
     useEffect(() => {
-        // do not change focus on the first render, as it should focus on the selected item
-        if (isInitialSectionListRender) {
+        // Avoid changing focus if the textInputValue remains unchanged.
+        if (prevTextInputValue === textInputValue || flattenedSections.allOptions.length === 0) {
             return;
         }
+        // Remove the focus if the search input is empty else focus on the first non disabled item
+        const newSelectedIndex = textInputValue !== '' ? 0 : -1;
```

Comment:
NAB, preferred a check rather than inverted check
```suggestion
        const newSelectedIndex = textInputValue === '' ? -1 : 0;
```